exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetScanBatchCreator.java

```diff
@@ -87,9 +87,6 @@ public ScanBatch getBatch(FragmentContext context, ParquetRowGroupScan rowGroupS
           newColumns.add(column);
         }
       }
-      if (newColumns.isEmpty()) {
```

Comment:
So, to clarify, the reason you removed the check for newColumns.isEmpty() is that if the column list is empty, the underlying ParquetRecordReader will handle it correctly by produce 1 default column (probably a NullableInt column) ? 
Was this check for isEmpty() only present in the Parquet scan ? or should other readers need modification too ? 

I think it would be good to add comments about how the NULL and empty column list are being handled by each data source.